File: 0070-climbing/main.js
// def climb_stairs_fast_fib2(n)
//   fibs = [1, 1]

//   0.upto(n) do |i|
//     fibs[i] ||= fibs[i - 1] + fibs[i - 2]
//   end

//   fibs[n]
// end


var climbStairs = function(n) {
  let fibs = [1, 1];
  for (i = 0; i <= n; i++) {
    fibs[i] ||= fibs[i - 1] + fibs[i - 2];
  }
  return fibs[n];
};

for (e = 1; e <=10; e++) {
  console.log(`climbStairs(${e})`, climbStairs(e))
}

File: 0070-climbing/main.go
package main

import ("fmt")

func climbStairs(n int) int {
  fibs := []int{1,1}
	for i:= 0; i <= n; i++ {
		if i >= len(fibs) {
			appFib := fibs[i - 1] + fibs[i - 2]
			fibs = append(fibs, appFib)
		}
	}
	return fibs[n]
}

func main () {
	for i := 1; i <= 10; i++ {
		fmt.Println(climbStairs(i))
	}
}